packages/db-kit/src/cli/menu/Menu.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState, useEffect } from "react";
+import { Box, Text } from "ink";
+import SelectInput from "ink-select-input";
+import type { Transaction, TransactionReceipt } from "web3-core";
+import type TruffleConfig from "@truffle/config";
+import type { Db, Project } from "@truffle/db";
+
+import type { Definitions, ModeInputProps, ModeName } from "./types";
+
+import * as DecodeAddress from "@truffle/db-kit/cli/decodeAddress";
+import * as DecodeTransaction from "@truffle/db-kit/cli/decodeTransaction";
+
+export interface Props {
+  config: TruffleConfig;
+  db: Db;
+  project: Project.Project;
+  onDone: () => void;
+}
+
+export type MenuModes = {
+  "decode-transaction": {
+    producesEffect: false;
+    rendersComponent: true;
+    props: Props & {
+      transactionHash: string;
+      transaction: Transaction;
+      receipt: TransactionReceipt;
+      addresses: string[];
+    };
+    inputPropName: "transactionHash" | "transaction" | "receipt" | "addresses";
+  };
+  "decode-address": {
+    producesEffect: false;
+    rendersComponent: true;
+    props: Props & {
+      address: string;
+    };
+    inputPropName: "address";
+  };
+  quit: {
+    rendersComponent: false;
+    producesEffect: true;
+    props: Props;
+  };
+};
+
+export const definitions: Definitions<MenuModes> = {
+  "decode-transaction": {
+    label: "Decode transaction",
+    propsInputComponent: props => {
+      return <DecodeTransaction.Inputs {...props} />;
+    },
```

Comment:
If you're just forwarding the props it should be the same as

```
propsInputComponent: DecodeTransaction.Inputs,
```

Reply:
oh, duh!

Reply:
I remember: I originally tried this, but I didn't know the bit about capitalization! Well, that's fixed now.